### ISSUE TITLE
Editing a data bag on Windows requests `client.pem`

### ISSUE THREAD

**Issue by emados** [thbishop/knife-solo_data_bag#35]
When I create an encrypted data bag (path to secret key is defined in `knife.rb`) using `knife solo` on Window 8.1, it gets created successfully.

``` bash
chef-repo $ knife solo data bag create sudoers auser -e vim
WARN: Auto inflation of JSON data is deprecated. Please pass in the class to inflate or use #edit_hash at C:/opscode/chefdk/embedded/lib/ruby/2.1.0/forwardable.rb:183:in `edit_data'
Created data_bag_item[auser]
```

But then, when I try to edit it, it fails and it asks for the `client.pem` to be present. It shouldn't do that as I'm using the `solo` sub command.

``` bash
chef-repo $ knife solo data bag edit sudoers auser -e vim
WARN: Failed to read the private key C:/chef/client.pem: #<Errno::ENOENT: No such file or directory @ rb_sysopen - C:/chef/client.pem>
ERROR: Your private key could not be loaded from C:/chef/client.pem
Check your configuration file and ensure that your private key is readable
```

Chef Development Kit Version: 0.15.15
chef-client version: 12.11.18
delivery version: 0.0.23 (bf89a6b776b55b89a46bbd57fcaa615c143a09a0)
berks version: 4.3.5
kitchen version: 1.10.0

The content of `.chef/knife.rb` is

``` ruby
cookbook_path    ["cookbooks", "site-cookbooks"]
node_path        "nodes"
role_path        "roles"
environment_path "environments"
data_bag_path    "data_bags"
encrypted_data_bag_secret ".chef/data_bag_key"

knife[:berkshelf_path] = "cookbooks"
Chef::Config[:ssl_verify_mode] = :verify_peer if defined? ::Chef
```

**Comment by ahukkanen** [comment]
This is not only Windows specific, also happens with Ubuntu (and I would guess in any environment).

Could you edit the title of this issue and remove the "Windows" part?

Workaround (while this is being fixed) for this is to use an older version of the Chef DK. It seems to work properly with `0.14.25`.

**Comment by ben-nz** [comment]
I can confirm the issue for both Windows 10 and Ubuntu 16.04, on ChefDK 0.16.28, and I can also confirm that reverting to 0.14.25 resolves the problem. Thanks for the workaround @ahukkanen

**Comment by Jamedjo** [comment]
Same problem using this with knife-solo gem. Similar workaround by temporarily adding `
gem 'chef', '= 12.10.24'` or `gem 'chef', github: 'chef/chef', tag: 'v12.10.53'` to my Gemfile to avoid [12.10.54](https://github.com/chef/chef/compare/f761b7f956f6b9a10301982c9c88167e7ade23ae...fb6e8bdc38ab125c6806cbe8765a60140c5f6673
) and later.

`gem 'chef', github: 'chef/chef', tag: 'v12.14.17'` didn't work, which was the latest version I could use without upgrading from ruby 2.1

**Comment by joeyates** [comment]
Hello @emados, if you put the following in `.chef/knife.rb` it should fix the issue:

```ruby
local_mode       true
```

**Comment by nachbar** [comment]
Same problem on a Mac, with a different error message:

```
WARN: Failed to read the private key /etc/chef/client.pem: #<Errno::ENOENT: No such file or directory @ rb_sysopen - /etc/chef/client.pem>
ERROR: Your private key could not be loaded from /etc/chef/client.pem
Check your configuration file and ensure that your private key is readable

```
Setting local_mode in .chef/knife.rb fixed this problem.  It might be worthwhile including this requirement in the Readme.

**Comment by joeyates** [comment]
@emados I've updated the README to indicate that with Chef 12.x, you need to add `local_mode true` to your `.chef/chef.rb`. Could you check if that solves your problem?

**Comment by joeyates** [comment]
I've released version 2.1.0 after merging #40.
Please check whether that fixes this issue.

**Comment by joeyates** [comment]
Closing as the issue should be fixed in 2.1.0
